src/components/shared/ButtonWithPanel/ArrowPanel.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+// This implements a panel with a "arrow" triangle graphic that points to the
+// button that triggers it.
+// Please do not use this component directly. This is used by ButtonWithPanel
+// to wrap the panel content.
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { ClosePanelContext } from './ClosePanelContext';
+
+import './ArrowPanel.css';
+
+type Props = {|
+  +onOpen: () => mixed,
+  +onClose: () => mixed,
+  +className?: string,
+  +children: React.Node,
+|};
+
+type State = {|
+  +open: boolean,
+  +isClosing: boolean,
+  +openGeneration: number,
+|};
+
+export class ArrowPanel extends React.PureComponent<Props, State> {
+  state = {
+    open: false,
+    isClosing: false,
+    openGeneration: 0,
+  };
+
+  open() {
+    if (this.state.open) {
+      return;
```

Comment:
codecov warning here is because the buttons are disabled in this test, so we can't reach this if block. I still added these if condition in case we want to remove the "disabled" attribute later.